src/cmdstan/command.hpp

```diff
@@ -644,50 +649,34 @@ int command(int argc, const char *argv[]) {
       throw std::invalid_argument(msg.str());
     }
 
-    size_t c_params_vec_size = 0;
-    size_t c_params_size = 0;
-    std::vector<double> c_params_r;
-    std::vector<size_t> dims_c_params_r;
-    if (!(cpars_file->is_default())) {
-      std::string c_fname(cpars_file->value());
-      std::ifstream c_stream(c_fname.c_str());
-
-      std::shared_ptr<stan::io::var_context> cpars_context
-          = get_var_context(c_fname);
-
-      c_params_r = (*cpars_context).vals_r("params_r");
-      dims_c_params_r = (*cpars_context).dims_r("params_r");
-
-      c_params_vec_size = dims_c_params_r.size() == 2 ? dims_c_params_r[0] : 1;
-      c_params_size = dims_c_params_r.size() == 2 ? dims_c_params_r[1]
-                                                  : dims_c_params_r[0];
-    }
-
-    if (c_params_size > 0 && c_params_size != param_names.size()) {
-      msg << "Incorrect number of constrained parameters provided! "
-             "Model has "
-          << param_names.size() << " parameters but " << c_params_size
-          << " were found.";
-      throw std::invalid_argument(msg.str());
-    }
-
+    int has_cpars = !cpars_file->is_default();
     // Store in single nested array to allow single loop for calc and print
-    size_t num_par_sets = c_params_vec_size + u_params_vec_size;
+    size_t num_par_sets = u_params_vec_size + has_cpars;
     std::vector<std::vector<double>> params_r_ind(num_par_sets);
-
-    // Use Map with inner stride to operate on all values from parameter set
-    using StrideT = Eigen::Stride<1, Eigen::Dynamic>;
     std::vector<int> dummy_params_i;
-    for (size_t i = 0; i < c_params_vec_size; i++) {
-      Eigen::Map<Eigen::VectorXd, 0, StrideT> map_r(
-          c_params_r.data() + i, c_params_size, StrideT(1, c_params_vec_size));
+    if (!cpars_file->is_default()) {
+      std::string cpars_filename(cpars_file->value());
 
-      stan::io::array_var_context context(param_names, map_r, param_dimss);
-      model.transform_inits(context, dummy_params_i, params_r_ind[i], &msg);
+      std::shared_ptr<stan::io::var_context> cpars_context
```

Comment:
[question]
Why is this getting turned into a pointer rather than just being declared as a `var_context`?

Reply:
Because the existing function for reading in parameters/data (`get_var_context()`) returns an `std::shared_ptr`